stripe.go

```diff
@@ -279,6 +284,33 @@ type BackendImplementation struct {
 	requestMetricsBuffer chan requestMetrics
 }
 
+type lastResponseSetterWrapper struct {
```

Comment:
consider specializing these to metric enquing 

metricsResponseSetter

Reply:
Why? Do you mean
```go
type metricsResponseSetter struct {
	LastResponseSetter
	backend *BackendImplementation
        params *Params
}

func (s metricsResponseSetter) SetLastResponse (response *APIResponse) {
	var usage []string
	if params != nil {
		usage = params.usage
	}
	backend.maybeEnqueueTelemetryMetrics(response.RequestID, response.duration, usage)
	s.LastResponseSetter.SetLastResponse(response)
}

func (s *BackendImplementation) CallRaw(method, path, key string, form *form.Values, params *Params, v LastResponseSetter) error {
	...
	s.Do(..., &responseSetter)
	responseSetter := lastResponseSetterWrapper{v, s, params}
}
```
I like reading the metrics-setting logic inline inside `CallRaw`, and more broadly I prefer general to specialized, at least for private methods/structs.

Reply:
>Why? 
Mostly to get rid of layers of indirection. A callback calling another callback ...

>Do you mean
Yep, that.

Reply:
I made this change on the infrastructure branch!